domehunter/__init__.py

```diff
@@ -31,52 +46,181 @@ class UnsupportedPythonError(Exception):
-            HOME_SENSOR_PIN_NUMBER = None
-
-        self.dome_status = "unknown"
-
+            # input 1 on automation hat
+            ENCODER_PIN_NUMBER = 26
+            # input 2 on the automation hat
+            HOME_SENSOR_PIN_NUMBER = 20
+            # relay 1 on automation hat
+            ROTATION_RELAY_PIN_NUMBER = 13
+            # relay 2 on automation hat
+            # no position for CW and nc for CCW?
+            DIRECTION_RELAY_PIN_NUMBER = 19
+            # set the timeout length variable to None for non testing mode
+            WAIT_TIMEOUT = None
+            # set a variable for bounce_time in seconds, this is just cool
+            # off period where the object will ignore additional (de)activation
+            BOUNCE_TIME = 0.1
+
+        if debug_lights:
+            # led_status is set with binary number, each zero/position sets the
+            # state of an LED, where 0 is off and 1 is on
+            self.led_status = 0b000000000000000000
+            sn3218.output([0x10] * 18)
+            sn3218.enable_leds(self.led_status)
+            sn3218.enable()
+            # create a dictionary of the LED name as the key and the sigit of
+            # the self.led_status binary int it corresponds to. This means we
+            # can convert the binary int to a string and use the index to
+            # change a 0 to 1 and vice versa and then convert back to a binary
+            # int. The updated self.led_status can then be sent to the LED
+            # controller.
+            self.led_lights_ind = {
+                'power': 2,
+                'comms': 3,
+                'warn': 4,
+                'input_1': 5,
+                'input_2': 6,
+                'input_3': 7,
+                'relay_3_nc': 8,
+                'relay_3_no': 9,
+                'relay_2_nc': 10,
+                'relay_2_no': 11,
+                'relay_1_nc': 12,
+                'relay_1_no': 13,
+                'output_3': 14,
+                'output_2': 15,
+                'output_1': 16,
+                'adc_3': 17,
+                'adc_2': 18,
+                'adc_1': 19
+            }
+
+        # initialize status and az as unknown, to ensure we have properly
+        # calibrated az
+        self.testing = testing
+        self.debug_lights = debug_lights
+        self._dome_status = "unknown"
+        self.dome_az = None
+
+        # create a instance variable to track the dome motor encoder ticks
         self.encoder_count = 0
-        self.encoder = DigitalInputDevice(ENCODER_PIN_NUMBER)
+        # bounce_time settings gives the time in seconds that the device will
+        # ignore additional activation signals
+        self.encoder = DigitalInputDevice(
+            ENCODER_PIN_NUMBER, bounce_time=BOUNCE_TIME)
         # _increment_count function to run when encoder is triggered
         self.encoder.when_activated = self._increment_count
+        # dummy value
+        self.az_per_tick = 1
```

Comment:
```suggestion
        self.az_per_tick = 1
```
You could set to `None` to ensure someone doesn't just start running this without calibrating it first. Or you could add a default `AZ_PER_TICK` as you do for the GPIO pins? If it isn't specified in e.g. the #22 yaml file upgrade, you could default to None so it requires a rotation calibration run?